Plutarch/Builtin.hs

```diff
@@ -245,7 +265,7 @@ instance PEq (PAsData a) where
   x #== y = punsafeBuiltin PLC.EqualsData # x # y
 
 instance (forall (s :: S). Coercible (a s) (Term s b), PIsData b) => PIsData (DerivePNewtype a b) where
-  pfromData x = pcon . DerivePNewtype $ ptypeOuter target
+  pfromData x = punsafeFrom target
```

Comment:
I changed this to use `punsafeFrom`, I assume this should be a safe usage here?

Reply:
why replace something safe with something unsafe?

Reply:
eh, mostly to remove the boilerplate a bit. I *assumed* they both have the same meaning in this case - but of course, I'm not entirely sure. Should this be changed back?

Reply:
Yeah they do, this is fine.